### PR TITLE
[msal-angular-v2] Merge to dev

### PR DESCRIPTION
**Pull request by jo-arroyo**
This PR merges `msal-angular-v2` to the dev branch, and includes:
- The v2 library replacing the v1 library in `/lib`
- Unit tests for the v2 library
- Additional v2 samples
  - angular 9 sample
  - angular 10 sample
  - angular 11 sample
  - angular 11 B2C sample
- Documentation for v2
  - Docs for v1 are preserved and separated out from v2 docs
  - Updated links in all docs, pointing to v1 on `msal-angular-v1` branch
- Change logs

The original v1 library is now found on the [`msal-angular-v1`](https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/msal-angular-v1/lib/msal-angular) branch.

### CODE REVIEW

**Review by jasonnutter — APPROVED** [review]

File: lib/msal-angular/docs/v1-docs/cors-api.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# CORS API usage in MSAL Angular v2
```
Comment (tnorling) [inline lib/msal-angular/docs/v1-docs/cors-api.md:1]: Should this be in v2 docs?

Comment (tnorling) [reply lib/msal-angular/docs/v1-docs/cors-api.md:1]: Oh I see this is also in v2. Is this also applicable to v1?

Comment (jo-arroyo) [reply lib/msal-angular/docs/v1-docs/cors-api.md:1]: Ah. It is applicable (the sample code below is different for v1 and v2), but I need to change the title. Thanks for catching that!

File: lib/msal-angular/src/msal.interceptor.ts [hunk 2]
```diff
@@ -3,61 +3,82 @@
...
 
 @Injectable()
 export class MsalInterceptor implements HttpInterceptor {
-    constructor(private auth: MsalService ,  private broadcastService: BroadcastService) {}
+    constructor(
+        @Inject(MSAL_INTERCEPTOR_CONFIG) private msalInterceptorConfig: MsalInterceptorConfiguration,
+        private authService: MsalService
+    ) {}
 
     intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
```
Comment (tnorling) [inline lib/msal-angular/src/msal.interceptor.ts:27]: We should address these linter warnings at some point after this PR

**Review by tnorling — APPROVED** [review]
Are the Angular 1 samples staying on dev or are they also being moved to the new branch?

File: lib/msal-angular/src/msal.interceptor.ts (hunk 2 above)
Comment (jo-arroyo) [reply lib/msal-angular/src/msal.interceptor.ts:27]: Yes, definitely!

File: lib/msal-angular/docs/v1-docs/cors-api.md (hunk 1 above)
Comment (jo-arroyo) [reply lib/msal-angular/docs/v1-docs/cors-api.md]: Changed.

**Comment by jo-arroyo** [comment]
> Are the Angular 1 samples staying on dev or are they also being moved to the new branch?

They will be on the new branch with msal-angular@1, but I think for the moment we will leave them on dev as well. In case of confusion, I added a README to the Angular 1 samples folder to make it clear they are for msal-angular@1, and that there is another folder of samples for msal-angular@2.

**Pull request merged**